tests/test_other.py

```diff
@@ -9906,3 +9907,18 @@ def test_relocatable_limited_exports(self):
     self.assertGreater(len(exports_linkable), 1000)
     self.assertIn('sendmsg', exports_linkable)
     self.assertNotIn('sendmsg', exports)
+
+  def test_nm_cache(self):
+    self.run_process([EMCC, path_from_root('tests', 'hello_world.c'), '-c'])
+    with env_modify({'EMCC_DEBUG': '1'}):
+      err = self.run_process([EMCC, 'hello_world.o'], stderr=PIPE).stderr
+      self.assertContained('llvm-nm', err)
+
+      # Second time linking the same object should used the cached nm result
+      err = self.run_process([EMCC, 'hello_world.o'], stderr=PIPE).stderr
+      self.assertNotContained('llvm-nm', err)
+
+      # Touch the object file and it should trigger nm once again.
+      Path('hello_world.o').touch(exist_ok=True)
```

Comment:
is there a chance of the test failing if the touch doesn't increment the time (if the resolution is seconds, say, and the test is faster than that)? not sure how best to solve that if it's an issue.

Reply:
Grepping for similar things, i see
```
tests/test_other.py-    # ensure some time passes before running the packager again so that if it does touch the
tests/test_other.py-    # js file it will end up with the different timestamp.
tests/test_other.py:    time.sleep(1.0)
```

Reply:
Yuck... can we wait and see if we get flakiness on this test before doing such a thing?

Reply:
Yes, not urgent.